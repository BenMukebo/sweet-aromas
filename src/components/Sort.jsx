import React, { useState } from 'react';
import { IconContext } from 'react-icons';
import { FiChevronDown } from 'react-icons/fi';

export default function Sort() {
  const [sortMethod, setSortMethod] = useState('None');
  const [showSort, setShowSort] = useState(false);

  const sortOptions = ['None', 'Name', 'Category'];
  const handleChangeSortMethod = (method) => {
    setSortMethod(method);
    setShowSort(false);
  };

  const handleToggleSort = () => {
    setShowSort(!showSort);
  };

  return (
    <div className="sort-main-wrapper d-flex flex-center flex-between">
      <span>sort by: </span>
      <a className="selected-filter-method d-flex flex-center" href="#f" onClick={handleToggleSort}>
        {sortMethod}
        <IconContext.Provider value={{ className: 'sort-arrow' }}>
          <FiChevronDown />
        </IconContext.Provider>
      </a>
      <ul className={showSort ? 'sort-method-list shown' : 'sort-method-list'}>
        {sortOptions.map((s) => (
          <li key={s}>
            <a href="#f" onClick={() => handleChangeSortMethod(s)}>
              {s}
            </a>
          </li>
        ))}
      </ul>
    </div>
  );
}
