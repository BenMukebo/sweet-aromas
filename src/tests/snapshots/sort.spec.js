import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Sort from '../../components/Sort';

describe('Sort', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Sort onChangeSortColumn={() => {}} activeColumn="None" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with sort column', () => {
    render(<Sort onChangeSortColumn={() => {}} activeColumn="Area" />);
    const sortColumn = screen.getAllByText(/Area/i);
    const element = sortColumn[0];
    expect(element).toBeInTheDocument();
  });

  it('should throw an error if looking for an unknown sort column', () => {
    render(<Sort onChangeSortColumn={() => {}} activeColumn="Area" />);
    expect(() => screen.getAllByText(/manger/i)).toThrow();
  });
});
