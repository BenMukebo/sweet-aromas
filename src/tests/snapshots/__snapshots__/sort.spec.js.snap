// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort should render correctly 1`] = `
<div
  className="sort-main-wrapper d-flex flex-center flex-between"
>
  <span>
    Sort By: 
  </span>
  <a
    className="selected-filter-method d-flex flex-center"
    href="#f"
    onClick={[Function]}
  >
    None
    <svg
      className="sort-arrow"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="6 9 12 15 18 9"
      />
    </svg>
  </a>
  <ul
    className="sort-method-list"
  >
    <li>
      <a
        data-sort="None"
        href="#f"
        onClick={[Function]}
        value="None"
      >
        None
      </a>
    </li>
    <li>
      <a
        data-sort="Name"
        href="#f"
        onClick={[Function]}
        value="Name"
      >
        Name
      </a>
    </li>
    <li>
      <a
        data-sort="Area"
        href="#f"
        onClick={[Function]}
        value="Area"
      >
        Area
      </a>
    </li>
  </ul>
</div>
`;
